stackslib/src/chainstate/nakamoto/test_signers.rs

```diff
@@ -124,18 +134,113 @@ impl Default for TestSigners {
+                    num_keys,
+                    threshold,
+                    &mut rng,
+                )
+            })
+            .collect();
+
+        // Generate an aggregate public key
+        let poly_commitments = match wsts::v2::test_helpers::dkg(&mut signer_parties, &mut rng) {
+            Ok(poly_commitments) => poly_commitments,
+            Err(secret_errors) => {
+                panic!("Got secret errors from DKG: {:?}", secret_errors);
+            }
+        };
+        let mut sig_aggregator = wsts::v2::Aggregator::new(num_keys, threshold);
+        sig_aggregator
+            .init(&poly_commitments)
+            .expect("aggregator init failed");
+        let aggregate_public_key = sig_aggregator.poly[0];
+        Self {
+            signer_parties,
+            aggregate_public_key,
+            poly_commitments,
+            num_keys,
+            threshold,
+            party_key_ids,
+            cycle: 0,
+            signer_keys,
+        }
+    }
+
+    /// Sign a Nakamoto block using [`Self::signer_keys`].
+    ///
+    /// N.B. If any of [`Self::signer_keys`] are not in the reward set, the resulting
+    /// signatures will be invalid. Use [`Self::sign_block_with_reward_set()`] to ensure
+    /// that any signer keys not in the reward set are not included.
     pub fn sign_nakamoto_block(&mut self, block: &mut NakamotoBlock, cycle: u64) {
         // Update the aggregate public key if the cycle has changed
         if self.cycle != cycle {
             self.generate_aggregate_key(cycle);
         }
 
-        let mut rng = rand_core::OsRng;
+        // TODO: epoch gate for aggregated signatures
+        // let signer_signature = self.sign_block_with_aggregate_key(&block);
+
+        let signer_signature = self.generate_block_signatures(&block);
+
+        test_debug!(
+            "Signed Nakamoto block {} with {} signatures (rc {})",
+            block.block_id(),
+            signer_signature.len(),
+            cycle
+        );
+        block.header.signer_signature = signer_signature;
+    }
+
+    /// Sign a NakamotoBlock and maintain the order
+    /// of the reward set signers in the resulting signatures.
+    ///
+    /// If any of [`Self::signer_keys`] are not in the reward set, their signatures
+    /// will be ignored.
+    pub fn sign_block_with_reward_set(&self, block: &mut NakamotoBlock, reward_set: &RewardSet) {
+        let signatures = self.generate_block_signatures(block);
+        let reordered_signatures = self.reorder_signatures(signatures, reward_set);
+        block.header.signer_signature = reordered_signatures;
+    }
+
+    /// Sign a Nakamoto block and generate a vec of signatures
+    fn generate_block_signatures(&self, block: &NakamotoBlock) -> Vec<MessageSignature> {
+        let msg = block.header.signer_signature_hash().0;
+        self.signer_keys
+            .iter()
+            .map(|key| key.sign(&msg).unwrap())
+            .collect::<Vec<MessageSignature>>()
```

Comment:
Nit: You ought to be able to omit the superfish operator here because the return type of this function will allow the Rust compiler to infer the type on its own.